extension/encoding/googlecloudlogentryencodingextension/internal/passthroughnlb/parser.go

```diff
@@ -0,0 +1,142 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package passthroughnlb contains utilities for parsing Google Cloud Passthrough External and Internal Network Load Balancer logs.
+package passthroughnlb // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/googlecloudlogentryencodingextension/internal/passthroughnlb"
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	gojson "github.com/goccy/go-json"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	semconv "go.opentelemetry.io/otel/semconv/v1.38.0"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/googlecloudlogentryencodingextension/internal/shared"
+)
+
+const (
+	// ConnectionsLogNameSuffix identifies load balancer connection logs in the logName field.
+	ConnectionsLogNameSuffix = "loadbalancing.googleapis.com%2Fflows"
+
+	externalLoadBalancerLogType = "type.googleapis.com/google.cloud.loadbalancing.type.ExternalNetworkLoadBalancerLogEntry"
+	internalLoadBalancerLogType = "type.googleapis.com/google.cloud.loadbalancing.type.InternalNetworkLoadBalancerLogEntry"
+
+	gcpPassthroughNLBPacketsStartTime = "gcp.load_balancing.passthrough_nlb.packets.start_time" // #nosec G101
+	gcpPassthroughNLBPacketsEndTime   = "gcp.load_balancing.passthrough_nlb.packets.end_time"
+	gcpPassthroughNLBBytesReceived    = "gcp.load_balancing.passthrough_nlb.bytes_received"
+	gcpPassthroughNLBBytesSent        = "gcp.load_balancing.passthrough_nlb.bytes_sent"
+	gcpPassthroughNLBPacketsReceived  = "gcp.load_balancing.passthrough_nlb.packets_received"
+	gcpPassthroughNLBPacketsSent      = "gcp.load_balancing.passthrough_nlb.packets_sent"
```

Comment:
@constanca-m do you think this should be `gcp.load_balancing.passthrough_nlb.packets.received` and `gcp.load_balancing.passthrough_nlb.packets.sent` to match the format of `.packets.start_time` / `.packets.end_time`?

Reply:
But then we have the inconsistency with `bytes_sent` and `bytes_received`